### PR TITLE
Add async mode to Radius Acct msgs

### PR DESCRIPTION
**Pull request by vijay-hirani**
This branch supports configuration for sending of radius messages in aync mode.

### CODE REVIEW

File: test/radius_SUITE.erl [hunk 1]
```diff
@@ -159,18 +159,19 @@ accounting_async() ->
...
-    set_async_mode(),
+    OrigApps = set_service_pars([{async, true}, {retries, 1}, {timeout, 5000}]),
     {ok, Session} = ergw_aaa_session_sup:new_session(self(),
 						     #{'Framed-IP-Address' => {10,10,10,10}}),
     ?equal(success, ergw_aaa_session:authenticate(Session, #{})),
     ?match({ok, _, _}, ergw_aaa_session:start(Session, #{}, [])),
     ?match({ok, _, _}, ergw_aaa_session:interim(Session, #{}, [])),
     ?match({ok, _, _}, ergw_aaa_session:stop(Session, #{}, [])),
-
+
```
Comment (RoadRunnr) [inline test/radius_SUITE.erl:169]: whitespace change?

Comment (vijay-hirani) [reply test/radius_SUITE.erl]: oops, removed it

File: src/ergw_aaa_radius.erl [hunk 2]
```diff
@@ -737,8 +747,10 @@ radius_accounting_opts() ->
...
 		       ?Event_Timestamp]).
 
-send_request(Req, Session, #{server := NAS} = Opts) ->
+send_request(Req, Session, #{server := NAS, retries := Retries,
+			     timeout := Timeout} = Opts) when is_tuple(NAS)->
     Id = maps:get('NAS-Identifier', Session, <<"NAS">>),
     RadiusClientOpts = [{client_name, maps:get(client_name, Opts, Id)},
 			{server_name, maps:get(server_name, Opts, Id)}],
-    eradius_client:send_request(NAS, Req, RadiusClientOpts).
+    eradius_client:send_request(NAS, Req, RadiusClientOpts ++
```
Comment (RoadRunnr) [inline src/ergw_aaa_radius.erl:755]: why the `++`? you could initialize the RadiusClientOpts with the values.

Comment (RoadRunnr) [inline src/ergw_aaa_radius.erl:755]: why `++`? You could just initialize `RadiusClientOpts` the all options.

File: test/config_SUITE.erl [hunk 3]
```diff
@@ -32,9 +32,11 @@
 	 {'Service-Type',          'Framed-User'}]).
 
 -define(RADIUS_AUTH_CONFIG,
-	[{server, {{127,0,0,1}, 1812, <<"secret">>}}]).
+	[{server, {{127,0,0,1}, 1812, <<"secret">>}},
+	 {retries, 3}, {timeout, 5000}]).
 -define(RADIUS_ACCT_CONFIG,
-	[{server,    {{0,0,0,0,0,0,0,1}, 1813, <<"secret">>}}]).
+	[{server,    {{0,0,0,0,0,0,0,1}, 1813, <<"secret">>}},
+	 {retries, 3}, {timeout, 5000}]).
```
Comment (RoadRunnr) [inline test/config_SUITE.erl:39]: there should be tests for the new options as well.

File: src/ergw_aaa_radius.erl (hunk 2 above)
Comment (vijay-hirani) [reply src/ergw_aaa_radius.erl:755]: Implemented.
the  async option is used here. 
https://github.com/travelping/ergw_aaa/blob/feature/add_radius_async_config/src/ergw_aaa_radius.erl#L143
Currently for the radius accounting stop msg it is always set to async by ergw.  need to configure async for accounting start msg in WUS.

Comment (vijay-hirani) [reply src/ergw_aaa_radius.erl:755]: implemented

**Pull request merged**